nixos/tests/boot.nix

```
@@ -42,7 +42,7 @@ let
 
   iso =
     (import ../lib/eval-config.nix {
-      inherit system;
+      inherit system pkgs;
       modules = [
```

- `pkgs` is a redundant argument we might want to get rid of; see comment in `eval-config.nix` - use the option instead
- It'd be good to make sure we're not accidentally ignoring any of the `nixpkgs.*` settings by making them read-only.
- `system` is also redundant, because it's only needed for creating a `pkgs`, which we already have.

```suggestion
      modules = [
        { nixpkgs.pkgs = pkgs; }
        ../modules/misc/nixpkgs/read-only.nix
```

(Same applies to the other `eval-config.nix` calls)

---

Alright, I've made that change

---

Removing system causes this:
```
       error: The option `nixpkgs.system' does not exist. Definition values:
       - In `/nix/store/fqjbkkwvb4h14bb4vl9pb28pja11bpcv-source/nixos/lib/eval-config.nix':
           {
             _type = "override";
             content = "aarch64-linux";
             priority = 1000;
           }
```

---

I guess that could be prevented by passing `system = null;` to `eval-config.nix`, but at that point we're adding too much boilerplate in my opinion. Let's postpone my proposal for when we have a nice and clean entrypoint. (in `nixos/lib/default.nix`?)

---

Yeah, that fixed it.